plugins/search-react/src/components/SearchAutocomplete/SearchAutocomplete.tsx

```diff
@@ -0,0 +1,202 @@
+/*
+ * Copyright 2022 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React, { ChangeEvent, ReactNode, useCallback, useMemo } from 'react';
+
+import {
+  CircularProgress,
+  ListItemIcon,
+  ListItemText,
+  ListItemTextProps,
+} from '@material-ui/core';
+import {
+  Value,
+  Autocomplete,
+  AutocompleteProps,
+  AutocompleteChangeDetails,
+  AutocompleteChangeReason,
+  AutocompleteRenderInputParams,
+} from '@material-ui/lab';
+
+import { SearchContextProvider, useSearch } from '../../context';
+import { SearchBar, SearchBarProps } from '../SearchBar';
+
+/**
+ * Props for {@link SearchAutocomplete}.
+ *
+ * @public
+ */
+export type SearchAutocompleteProps<
+  T,
+  Multiple extends boolean | undefined,
+  DisableClearable extends boolean | undefined,
+  FreeSolo extends boolean | undefined,
```

Comment:
This isn't super clear to me, can you explain why this is needed? is it something coming from the mui autocomplete component?

Reply:
The Material UI Autocomplete props require these generics.

Reply:
got it, maaaaybe worth a small clarifying comment pointing to the component?

Reply:
I simplified the types. We don't have the option to select multiple terms, we always want to be able to clear the input value, and our default search is free solo, so now we only infer the option type. Does it work better?